lib/pymedphys/docs/contrib/info/release-guide.md

```diff
@@ -0,0 +1,111 @@
+# pymedphys Release Procedure
+
+## Determine next release version
+
+pymedphys uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html) in the format `MAJOR.MINOR.PATCH`, the next release number should typically be `MAJOR.MINOR+1.PATCH`. For example, if the previous release was `0.38.0`, the upcoming release will be `0.39.0`.
+
+In instances where the only changes since the last release are bug fixes and none of the pymedphys API has changed, you should increment the `PATCH` value: `MAJOR.MINOR.PATCH+1`
+
+In the remainder of this guide this version code will be referred to as `VERSION`.
+
+## Create a branch to prepare the release
+
+Create a branch named `VERSION-release-prep`:
+
+```bash
+git checkout -b VERSION-release-prep
+git push --set-upstream origin 0.39.0-release-prep
+```
+
+## Update version in pyproject.toml
+
+Update the version code near the top of the file:
+
+```toml
+[tool.poetry]
+name = "pymedphys"
+version = "VERSION"
+readme = "README.rst"
+...
+```
+
+Then run poetry update as well as propagate:
+
+```bash
+poetry update
+poetry run pymedphys propagate
+```
+
+## Update CHANGELOG
+
+Amend the `CHANGELOG.md` file to describe the changes since the last release. Insert information for this release near the top of the file and populate the sections as appropriate (remove unused sections):
+
+```markdown
+<!-- markdownlint-disable MD024 MD039 -->
+
+# Release Notes
+
+All notable changes to are documented here.
+
+This project adheres to
+[Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [VERSION]
+
+### News around this release
+
+### Breaking changes
+
+### New features and enhancements
+
+### Bug Fixes
+
+...
+```
+
+To help determine what has changed since the last release, you can inspect the [merged pull requests](https://github.com/pymedphys/pymedphys/pulls?q=is%3Apr+is%3Amerged) during that time frame.
+
+## Create release pull request
+
+Commit all changes to your release branch and push to GitHub. Then create a Pull Request of this branch into main. This will run the CI tests. Before finalising the release, ensure all tests are passing.
+
+### Troubleshooting issues
+
+When preparing a release you will typically see certain steps of the CI failing. This is usually due to dependencies having been upgraded in the time since the last release and certain code within pymedphys is no longer compatible with the new versions.
+
+To confirm this is indeed the problem, you can set an upper pin for these libraries (to constrain them to a previous version). By doing this you should hopefully see the CI pass.
+
+In most instances, releasing pymedphys with an upper pin isn't appropriate, as this would be too restrictive for pymedphys users. Unfortunately there isn't a general solution to resolving these issues. You will need to remove the upper pins one-by-one and adjust the code in pymedphys to support the new version of the dependencies.
```

Comment:
```suggestion
In most instances, releasing pymedphys with an upper pin isn't appropriate. `pymedphys` is primarily a _library_ and upper pins introduce future obsolescnece risk. Unfortunately there isn't a general solution to resolving these issues. You will need to remove the upper pins one-by-one and adjust the code in pymedphys to support the new version of the dependencies.
```

Reply:
Less so obsolescence... more so... "dependency hell" for those trying to install the library in combination with other libraries that have their own restrictions.

Reply:
Sure, "primarily" misses the point that this is a library release.

I do think upper pins technically introduce obsolescence, which at least worsens in the "future", but I can see that perhaps it's not the primary point. Feel free to suggest alternate wording. I just thought it best to give a reason.

I had in mind the point that, at some time in the future, a pymedphys dependency will likely introduce lower-pins that conflict with another (cross-)dependency's upper-pin that we've introduced. So, "obsolescence" and "future".